Mlem/Views/Tabs/Posts/Components/Shared/Links/User Profile Link.swift

```diff
@@ -26,24 +43,117 @@ struct UserProfileLink: View
                     }
                 }
                 
-                Text(user.name)
+                /* Disabled icons for now */
+                if let flair = calculateFlair() {
+                    if let flairSystemIcon = flair.systemIcon {
+                        Image(systemName: flairSystemIcon).foregroundColor(flair.color)
+                    }
+                }
+                
+                
+                // User display name if one exists
+                Text(user.displayName ?? user.name)
                     .minimumScaleFactor(0.01)
                     .lineLimit(1)
             }
-            .if(user.admin)
+            .if(calculateFlair() != nil)
             { viewProxy in
                 viewProxy
-                    .foregroundColor(.red)
+                    .foregroundColor(calculateFlair()!.color).bold()
             }
-            .if(user.botAccount == true)
-            { viewProxy in
-                viewProxy
-                    .foregroundColor(.indigo)
+        }
+    }
+    
+    struct UserProfileLinkFlair {
+        var color: Color
+        var systemIcon: String? = nil
+    }
+    
+    private func calculateFlair() -> UserProfileLinkFlair? {
```

Comment:
it could be worth making the `UserProfileLinkFlair` a standalone struct with an initialiser that takes the parameters it requires, the initialiser can then do much the same logic as `calculateFlair` and return the correct value?

if it was stand-alone it'd make re-using it elsewhere easier too 🙏

Reply:
I just tweaked the name, but this method actually calculates both the color AND the icon ("flair") so I'm not sure if spinning this off to its own view makes much sense.  The `UserProfileLink` its self is a pretty stand-alone, reused view though.